### PR TITLE
std.encoding marked unittests and functions @safe pure nothrow

### PR DESCRIPTION
**Pull request by burner**

### CODE REVIEW

File: std/encoding.d [hunk 1]
```diff
@@ -748,17 +748,17 @@ template EncoderInstance(CharType : AsciiChar)
...
     alias EString = AsciiString;
 
-    @property string encodingName()
+    @property string encodingName() @safe pure nothrow
     {
         return "ASCII";
     }
 
-    bool canEncode(dchar c)
+    bool canEncode(dchar c) @safe pure nothrow
```
Comment (JackStouffer) [inline std/encoding.d]: `@nogc`?

Comment (burner) [inline std/encoding.d]: done

File: std/encoding.d [hunk 2]
```diff
@@ -2373,12 +2373,12 @@ abstract class EncodingScheme
         /**
          * Returns the standard name of the encoding scheme
          */
-        abstract override string toString();
+        abstract override string toString() @safe pure nothrow @nogc;
```
Comment (JackStouffer) [inline std/encoding.d]: Wouldn't this require that all classes that inherit from this also have their `toString` method `@safe pure nothrow @nogc`? That would be a backwards compatibility breaking change.

Comment (burner) [inline std/encoding.d]: my bad

File: std/encoding.d [hunk 3]
```diff
@@ -2660,7 +2660,7 @@ class EncodingSchemeASCII : EncodingScheme
             ];
         }
 
-        override string toString()
+        override string toString() @safe pure nothrow
```
Comment (JackStouffer) [inline std/encoding.d]: `@nogc`

File: std/encoding.d [hunk 4]
```diff
@@ -2743,7 +2743,7 @@ class EncodingSchemeLatin1 : EncodingScheme
...
```
Comment (JackStouffer) [inline std/encoding.d]: ditto

File: std/encoding.d [hunk 5]
```diff
@@ -2818,7 +2818,7 @@ class EncodingSchemeLatin2 : EncodingScheme
...
```
Comment (JackStouffer) [inline std/encoding.d]: ditto

File: std/encoding.d [hunk 6]
```diff
@@ -2877,15 +2877,15 @@ class EncodingSchemeWindows1250 : EncodingScheme
...
+        override string[] names() @safe pure nothrow
         {
             return
             [
                 "windows-1250"
             ];
         }
 
-        override string toString()
+        override string toString() @safe pure nothrow
```
Comment (JackStouffer) [inline std/encoding.d]: ditto

**Comment by JackStouffer** [comment]
And elsewhere

**Comment by JackStouffer** [comment]
LGTM

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thanks

**Pull request merged**